packages/clerk-js/src/core/clerk.ts

```diff
@@ -1039,8 +1049,50 @@ export default class Clerk implements ClerkInterface {
     await this.navigate(this.#buildPrimarySyncUrl());
   };
 
+  #shouldSyncWithPrimaryDev = (): boolean => {
+    if (this.isSatellite && !this.#options.signInUrl && this.#instanceType === 'development') {
+      clerkMissingSignInUrlAsSatellite();
+      return false;
+    }
+
+    return this.isSatellite && this.#instanceType === 'development';
+  };
+
+  #syncWithPrimaryDev = async () => {
+    await this.navigate(this.#buildPrimarySyncUrlDev());
+  };
+
+  #buildPrimarySyncUrlDev = (): string => {
+    const primarySyncUrl = new URL(`/`, this.#options.signInUrl);
+    primarySyncUrl?.searchParams.append(CLERK_SYNCING, 'true');
+    primarySyncUrl?.searchParams.append('redirect_url', window.location.href);
+
+    return primarySyncUrl?.toString() || '';
+  };
+
+  #shouldRedirectToSatelliteDev = (): boolean => {
+    if (this.isSatellite) {
+      return false;
+    }
+
+    return this.#isRequestingSyncing();
+  };
+
   #loadInStandardBrowser = async (): Promise<boolean> => {
-    if (this.#shouldSyncWithPrimary()) {
+    this.#devBrowserHandler = createDevBrowserHandler({
+      frontendApi: this.frontendApi,
+      fapiClient: this.#fapiClient,
+    });
+
+    if (this.#hasJustSynced()) {
```

Comment:
⛔ @panteliselef as the logic evolves, I'd suggest moving all the multidomain-related code in `syncWithPrimary` method for both dev and production instances.

Reply:
I will refactor this :)